cloudpickle/cloudpickle.py

```diff
@@ -676,10 +744,16 @@ def save_function_tuple(self, func):
         save(_fill_function)  # skeleton function updater
         write(pickle.MARK)    # beginning of tuple that _fill_function expects
 
-        self._save_subimports(
+        submodules = _find_loaded_submodules(
             code,
             itertools.chain(f_globals.values(), closure_values or ()),
         )
+        for s in submodules:
+            # ensure that subimport s is loaded at unpickling time
+            self.save(s)
+            # then discards the reference to it
+            self.write(pickle.POP)
```

Comment:
> Instead of doing this, why not use the same strategy as done in the cloudpickle_fast version, that is passing references to submodules in function globals that get discarded by the reconstructor function?

Agreed. The less crazy low-level pickle instructions, the better.

Reply:
This point wasn't resolved right?

Reply:
No. You're fine with doing that in this PR?

Reply:
Not necessarily. We can merge this at it seems to be working fine and we can refactor the non-C version in a dedicated PR.

Reply:
Ok you did it :)